poppy/poppy_core.py

```diff
@@ -34,18 +33,22 @@
     _FFTW_AVAILABLE = False
 
 # internal constants for types of plane
-_PUPIL = 'PUPIL'
-_IMAGE = 'IMAGE'
-_DETECTOR = 'DETECTOR' # specialized type of image plane.
-_ROTATION = 'ROTATION' # not a real optic, just a coordinate transform
-#_typestrs = {0:'', 1:'Pupil plane', 2:'Image plane', 3:'Detector', 4:'Rotation', 'PUPIL':'Pupil plane','IMAGE':'Image plane'}
+class PlaneType(enum.Enum):
+    unspecified = 0
```

Comment:
None could work, yes. It's used in `FresnelWavefront.apply_lens_power()`, to let you stick optics in the system that don't affect or don't care what the plane type is.   Its effect is "pass the `wavefront` through this optic, leaving the `wavefront.planetype` unchanged". 

The original motivating use case is in the Hubble example:
 - Circular Aperture in a pupil plane
 - primary mirror (*which is still a pupil plane*)
 - secondary mirror (intermediate plane)
 - image plane

In the earlier version of this code (e.g. your https://gist.github.com/josePhoenix/462096e25fff10267a75) note that the lenses had to have a plane type explicitly assigned. In my refactored code you can just stick a lens there without specifying the plane and it works as desired. 

More broadly, all of the "pupil plane" optics from `optics.py` can potentially be stuck at intermediate planes as well as pupils, or even images now. E.g. the ACS coronagraph occulting spot is in the aberrated beam near but not precisely in focus, or the WFIRST PIAACMC design uses a series of Lyot stops at intermediate planes.  So I think it's useful to be able to say "just let the light go through this without relabeling what plane type the wavefront is at."

More fundamentally, the entire `planetype` paradigm is not a great fit for Fresnel because the boundaries are not sharp. It's kept for compatibility with the rest of poppy of course. But if we were to make a standalone Fresnel code I would advocate getting rid of the entire notion of `planetype` and just having wavefronts with a coordinates toggle between physical and angular units.